src/server/graphql/models/Invitation/invitationMutation.js

```diff
@@ -23,9 +25,75 @@ export default {
       }
     },
     async resolve(source, {invitees, teamId}, {authToken}) {
+      const r = getRethink();
       requireSUOrTeamMember(authToken, teamId);
+      const now = Date.now();
+      // don't let them invite the same person twice
+      const emails = invitees.map(invitee => invitee.email);
+      const usedEmails = await r.table('Invitation')
+        .getAll(r.args(emails), {index: 'email'})
+        .filter(r.row('tokenExpiration').ge(r.epochTime(now)))('email')
+        .coerceTo('array')
+        .do((inviteEmails) => {
+          return {
+            invites: inviteEmails,
+            teamMembers: r.table('TeamMember')
+              .getAll(teamId, {index: 'teamId'})('email')
+              .coerceTo('array')
+          };
+        });
+      const alreadyInvited = invitees
+        .map((i) => i.email)
+        .filter((email) => usedEmails.invites.includes(email))
+        .join(', ');
+      if (alreadyInvited) {
+        throw errorObj({
+          _error: `${alreadyInvited.join(', ')} already invited`,
+          type: 'alreadyInvited'
+        });
+      }
+      const alreadyTeamMember = invitees
+        .map((i) => i.email)
+        .filter((email) => usedEmails.teamMembers.includes(email))
+        .join(', ');
+      if (alreadyTeamMember) {
+        throw errorObj({
+          _error: `${alreadyTeamMember.join(', ')} already on the team`,
+          type: 'alreadyTeamMember'
+        });
+      }
       asyncInviteTeam(authToken, teamId, invitees);
       return true;
     }
+  },
+  cancelInvite: {
+    type: GraphQLBoolean,
+    description: 'Cancel an invitation',
+    args: {
+      inviteId: {
+        type: new GraphQLNonNull(GraphQLID),
+        description: 'The id of the invitation'
+      },
+    },
+    async resolve(source, {inviteId}, {authToken}) {
+      await cancelInvitation(authToken, inviteId);
+      return true;
+    }
+  },
+  resendInvite: {
```

Comment:
they would if they typed in the email as if it were a new invitee. however, if you scroll down to the invitee & click "resend" then it will invalidate the old token and send a new email with a new token.